app/services/exchange_deleter.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class ExchangeDeleter
+  def delete(child, parent)
+    active_upcoming_order_cycles = OrderCycle.active.pluck(:id) + OrderCycle.upcoming.pluck(:id)
```

Comment:
So if I change the date of an inactive order cycle then I could make it active and still have those products in there? :thinking:

Reply:
Good point. I could add code that would delete the exchanges in that case, but we wouldn't be able to see the full order cycle history.

Or just destroy all exchanges even if they are expired. However, in that case we won't be able to access the history of the order cycle.

I don't know enough about the use cases to know which would be the better solution.

Reply:
Hm, we have that history problem already. For your PR, we should remove the items from all order cycles.

One common workflow is to clone a closed order cycle and set new dates to start a new one. Then it would make sense if any revoked products were already removed from the old order cycle.

If we need the feature of knowing which products were offered through an order cycle in the past then we need to implement that separately.